src/browserlib/extract-events.mjs

```diff
@@ -0,0 +1,278 @@
+	  if (el && el.tagName === "A" && el.getAttribute("href").startsWith("https:")) {
+	    // we skip when we hit a link pointing to an external spec
+	    // (this is needed since the HTML spec table includes
+	    // links to pointer events)
+	    return;
+	  }
+	  if (!el) {
+	    el = eventEl.querySelector("code");
+	  }
+	  if (el) {
+	    if (el.tagName === "DFN" && el.id) {
+	      event.href = href(el);
+	    } else if (el.tagName === "A") {
+	      event.href = href(document.getElementById(el.getAttribute("href").slice(1)));
+	    }
+	    event.src = { format: "summary table", href: href(el.closest('*[id]')) };
+	    event.type = eventEl.textContent.trim();
+	    event.targets = fromEventElementToTargetInterfaces(eventEl.querySelector("dfn,a[href^='#']"));
+	    if (bubblingInfoColumn >= 0) {
+	      event.bubbles = tr.querySelector(`td:nth-child(${bubblingInfoColumn + 1})`)?.textContent?.trim() === "Yes";
+	    }
+	    if (interfaceColumn >= 0) {
+	      event.interface = tr.querySelector(`td:nth-child(${interfaceColumn + 1}) a`)?.textContent ?? tr.querySelector(`td:nth-child(${interfaceColumn + 1}) code`)?.textContent;
+	    }
+	    events.push(event);
+	  }
+	});
+      } else if (table.className === "event-definition") {
+	hasStructuredData = true;
+	// Format used e.g. in uievents
+	const eventName = table.querySelector("tbody tr:first-child td:nth-child(2)")?.textContent.trim();
+	let iface = table.querySelector("tbody tr:nth-child(2) td:nth-child(2)")?.textContent.trim();
+	let bubbles = table.querySelector("tbody tr:nth-child(4) td:nth-child(2)")?.textContent.trim() === "Yes";
+	// Prose description, we skip it
+	if (iface.match(/\s/)) {
+	  iface = null;
+	}
+	let targets = table.querySelector("tbody tr:nth-child(5) td:nth-child(2)")?.textContent.split(",").map(t => t.trim());
+	if (targets && targets.find(t => t.match(/\s/))) {
+	  // Prose description, skip it
+	  targets = null;
+	}
+	if (eventName) {
+	  events.push({type: eventName, interface: iface, targets, bubbles, src: { format: "definition table", href: href(table.closest('*[id]')) }, href: href(table.closest('*[id]')) });
+	}
+      } else if (table.className === "def") {
+	// Used in https://drafts.csswg.org/css-nav-1/
+	const rowHeadings = [...table.querySelectorAll("tbody th")];
+	if (rowHeadings.find(th => th.textContent.trim() === "Bubbles")) {
+	  const eventTypeRow = [...table.querySelectorAll("tbody th")].findIndex(n => n.textContent.trim().match(/^type/i));
+	  const bubblingInfoRow = [...table.querySelectorAll("tbody th")].findIndex(n => n.textContent.trim() === "Bubbles");
+	  const interfaceRow = [...table.querySelectorAll("tbody th")].findIndex(n => n.textContent.trim().match(/^interface/i));
+	  const eventName = table.querySelector(`tr:nth-child(${eventTypeRow + 1}) td:nth-child(2)`)?.textContent?.trim();
+	  const bubbles = table.querySelector(`tr:nth-child(${bubblingInfoRow + 1}) td:nth-child(2)`)?.textContent?.trim() === "Yes";
```

Comment:
This does not distinguish between the case where we have the "bubbles" information and we know the even does not bubble and the case where we don't have the "bubbles" information and decide that it does not bubble. Shouldn't it somehow? For instance by setting `bubbles` to null or undefined when we don't know as done further below, although that's a bit ugly, or by using something else than a boolean to represent the info?